src/angular-modal-service.js

```diff
@@ -183,6 +183,10 @@ module.provider('ModalService', function ModalServiceProvider() {
                             modal.close = closeDeferred.promise;
                             modal.closed = closedDeferred.promise;
 
+                            if (angular.isFunction(modal.controller.$onInit)) {
```

Comment:
This is excellent - but can I suggest a comment here to note that the `$onInit` function is only present in Angular 1.7, so we must check for it before firing it? Might make it clearer for anyone checking this later!